### PR TITLE
feat: Added custom route for NAT gateway

### PR DESCRIPTION
**Pull request by piersf**
## Description

When having a VPC with public and private subnets, there are cases where traffic from the private subnet that is destined for `0.0.0.0/0` does not need to go to the NAT Gateway but instead, it should be sent to another destination (e.g., transit gateway). And then only specific traffic that is destined for the internet (e.g., 8.8.8.8/32) should be sent to the NAT Gateway.

Therefore, this change adds a new conditional variable that enables us to pass a custom destination CIDR block for the NAT Gateway.

## Motivation and Context

When having a VPC with public and private subnets, there are cases where traffic from the private subnet that is destined for `0.0.0.0/0` does not need to go to the NAT Gateway but instead, it should be sent to another destination (e.g., transit gateway). And then only specific traffic that is destined for the internet (e.g., 8.8.8.8/32) should be sent to the NAT Gateway.

## Breaking Changes

No breaking changes were noticed.

## How Has This Been Tested?
- [x] I have tested and validated these changes using one or more of the provided `examples/*` projects

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -298,6 +298,12 @@ variable "enable_nat_gateway" {
   default     = false
 }
 
+variable "private_nat_gateway_destination_route" {
+  description = "Used to pass a custom destination route for private NAT Gateway. If not specified, the default 0.0.0.0/0 is used as a destination route."
+  type        = string
+  default     = null
```
[the remark this answers is not in the document]
Comment (piersf) [reply variables.tf:304]: Absolutely, yes. I just took a conditional approach. Will change that now.

[the remark this answers is not in the document]
Comment (piersf) [reply variables.tf]: Corrected. If that works for you, I can resolve this conversation.

File: variables.tf [hunk 2]
```diff
@@ -298,6 +298,12 @@ variable "enable_nat_gateway" {
   default     = false
 }
 
+variable "private_nat_gateway_destination_route" {
```
[the remark this answers is not in the document]
Comment (piersf) [reply variables.tf]: This is now fixed.

**Comment by piersf** [comment]
> @piersf Please fix the docs (run `pre-commit run -a` and commit changes) and I will merge it.

@antonbabenko  silly question but how do I run `pre-commit run -a`? That gives me a `command not found` on my machine as obviously, I don't have such a package/tool installed.

**Comment by antonbabenko** [comment]
You need to install it - https://pre-commit.com/#installation and `terraform-docs` (see [github workflow config](https://github.com/terraform-aws-modules/terraform-aws-vpc/blob/master/.github/workflows/pre-commit.yml) for more).

Just let me know if you want me to fix this and release it.

**Comment by piersf** [comment]
> Just let me know if you want me to fix this and release it.

If you can, I'd really appreciate it. I don't have many of these prerequisites installed like `terraform-docs` and `tflint` and will take me some time to install them due to policies on the machine.

**Comment by antonbabenko** [comment]
This PR is included in version 3.12.0 :tada:

**Comment by antonbabenko** [comment]
No problems, I've just fixed it. Thank you for your contribution!

**Pull request merged**